### ISSUE TITLE
[BUG] : Compo Doc does not generate the documentation for project

### ISSUE THREAD

**Issue by mg-arrow** [compodoc/compodoc#82]
##### **Overview of the issue**

We have very large angular 2 typescript based application installed compodoc from npm and tried to create documentation . Process started but finished in the middle without any errors.

##### **Operating System, Node.js, npm, compodoc version(s)**

Node 4.2.4, latest version of compodoc, windows 7

##### **Angular configuration, a `package.json` file in the root folder**

[package-json.txt](https://github.com/compodoc/compodoc/files/720102/package-json.txt)

##### **Compodoc installed globally or locally ?**

installed globally

##### **Motivation for or Use Case**

no documentation generated, no error is generated

##### **Reproduce the error**

No sure how you can reproduce it because I would not be able to share the whole project

##### **Related issues**

Have not seen it before before

**Comment by vogloblinsky** [comment]
can you provide a screenshot of your terminal ?
what is the last log info ?

**Comment by mg-arrow** [comment]
Yes, sure .. the processing stops on one of the ts files.

Attached screenshots
[error.docx](https://github.com/compodoc/compodoc/files/720671/error.docx)

**Comment by vogloblinsky** [comment]
@mg-arrow can you send me an email please ? (Check my profile)

**Comment by zpydee** [comment]
i'm not sure if this is related to the same issue, but I have experienced situations wherein `compodoc` is also not outputting documentation for some of my `injectables`.

For example, this is an extract of one of my services:

```javascript
/**
 * The `AuthService` handles all app logic related to user authentication. This includes:
 * 
 * - observable and non-observable properties related to tracking authentication status.
 * - interactions with `components` that need to know about authentication status
 * - app `routing` related to authentication status
 * - communication with server end-points for user registration and authentication
 */
@Injectable()
export class AuthService {

  /** boolean indicating whether or not the app requires users to be authenticated to access main features  */
  requireLogin: boolean = true;
  /** `observable<boolean>` tracking whether or not a user is signed in. Used within various `components` to show/hide elements, as well as within `AuthGuardService` for protected `routes` */
  isLoggedIn$: Subject<boolean> = new Subject();
  /** stores the desired `url` that should be redirected to once a user has been authenticated (for purposes of deep links) */
  redirectUrl: string;
  /** `observable<boolean> that can be updated within `brochure components` to easily control if sign-in and sign-up buttons should be displayed. */
  showAuthButtons$: Subject<boolean> = new Subject();
  /** state variable that tracks the currently signed-in user */
  currentUser: User;
  /** state variable that tracks the currently signed-in user's authentication token as supplied by the server upon sign-in  */
  authToken: string;

  /**
   * - injects the `AuthServiceConfig`, specifically `requireLogin` to indicate whether or not the application requires authentication in order to access main app features.
   * - injects required dependencies as local private properties, including:
   *  - `Router` for navigation to app routes
   *  - `HttpService` for calls to server
   *  - `UserService` for interactions with the `User` model during authentication processes
   *  - `OrganisationService` for interactions with the `Organisations model during authentication processes
   * - Checks to see if `currentUser` and `authToken` have already been set in `local-storage` and if so, gets these and assigns them to their associated state variables.
   */
  constructor(
    @Optional() config: AuthServiceConfig,
    private router: Router,
    private httpService: HttpService,
    private userService: UserService,
    private organisationService: OrganisationService
  ) {
    if (config) {
      this.requireLogin = typeof config.requireLogin === 'undefined' ? true : config.requireLogin;
    }
    this.currentUser = JSON.parse(localStorage.getItem('current-user'));
    this.authToken = localStorage.getItem('auth-token');
  }


  /** updates the `showAuthButtons$` observable */
  toggleShowAuthButtons(show: boolean): void {
    this.showAuthButtons$.next(show);
  }
```

this is the output:

<img width="718" alt="screen shot 2017-01-21 at 4 03 21 pm" src="https://cloud.githubusercontent.com/assets/6129162/22174934/2c7d0a04-dff3-11e6-8858-770421b8985f.png">

there is clearly a lot missing...

**Comment by vogloblinsky** [comment]
Closed by version 0.0.36
